presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -518,6 +520,22 @@ public Boolean visitParameter(Parameter node, Void context)
             return process(parameters.get(node.getPosition()), context);
         }
 
+        public Boolean visitGroupingOperation(GroupingOperation node, Void context)
+        {
+            List<Expression> argumentsNotInGroupBy = node.getGroupingColumns().stream()
+                    .filter((argument) -> !expressions.contains(argument))
```

Comment:
Good catch! I think this is the only comment I won't address in this iteration. I'll rebase and squash all the changes I just made so I can merge this to the sprint branch and our release branch. We should then talk about the best way to address this. I had a brief look at your PR and it's probably best for me to re-use what you've written instead of implementing something from scratch.